### PR TITLE
private-cloud cmk-encryption commands, identity commands, and test cases

### PR DESCRIPTION
**Pull request by jonathanhe-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
vmware

File: src/vmware/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Release History
 
+## 4.0.3 (2022-07)
+
+- Add `az vmware private-cloud enable-cmk-encryption`
+- Add `az vmware private-cloud disable-cmk-encryption`
+- Deprecate `az vmware private-cloud add-cmk-encryption`
+- Deprecate `az vmware private-cloud delete-cmk-encryption`
+- Removed `--mi-system-assigned` parameter from `az vmware private-cloud identity assign`
```
[the remark this answers is not in the document]
Comment (cataggar) [reply src/vmware/HISTORY.md:9]: Please deprecate `az vmware private-cloud identity` command group.

File: src/vmware/HISTORY.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+- Renamed `--mi-system-assigned` parameter from `az vmware private-cloud create` to `--identity`
```
[the remark this answers is not in the document]
Comment (cataggar) [reply src/vmware/HISTORY.md:10]: Please also add `--identity` `az vmware private-cloud update`.

File: src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py [hunk 3]
```diff
@@ -96,13 +96,19 @@ def test_vmware(self):
...
+        self.cmd('vmware private-cloud identity assign -g {rg} -c {privatecloud}')
 
         # remove managed identity
         self.cmd('vmware private-cloud identity remove -g {rg} -c {privatecloud}')
 
         # show managed identity
         self.cmd('vmware private-cloud identity show -g {rg} -c {privatecloud}')
 
+        # enable cmk encryption
+        self.cmd('az vmware private-cloud enable-cmk-encryption -c {privatecloud} -g {rg} --enc-kv-key-name test-key-name --enc-kv-key-version 1 --enc-kv-url test-url')
```
Comment (cataggar) [inline src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py:108]: Let's get the CMK team to review these argument names.

**Review by cataggar — CHANGES_REQUESTED** [review]

File: src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py (hunk 3 above)
Comment (cataggar) [reply src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py:99]: Rahi approved.

**Review by cataggar — CHANGES_REQUESTED** [review]

File: src/vmware/azext_vmware/_params.py [hunk 4]
```diff
@@ -39,7 +44,7 @@ def load_arguments(self, _):
         c.argument('nsxt_password', help='NSX-T Manager password.')
         c.argument('accept_eula', help='Accept the end-user license agreement without prompting.')
         c.argument('network_block', help='A subnet at least of size /22. Make sure the CIDR format is conformed to (A.B.C.D/X) where A,B,C,D are between 0 and 255, and X is between 0 and 22.')
-        c.argument('mi_system_assigned', help='Enable a system assigned identity.')
+        c.argument('mi_system_assigned', help=clean_help(ResourceIdentityType.__doc__), arg_type=get_enum_type([identityType.value for identityType in ResourceIdentityType]), deprecate_info=c.deprecate(redirect='--identity', hide=True))
```
Comment (cataggar) [inline src/vmware/azext_vmware/_params.py:47]: Works for me. Let's see if @zhoxing-ms knows of a better way.

**Review by cataggar — APPROVED** [review]

File: src/vmware/azext_vmware/_params.py (hunk 4 above)
Comment (zhoxing-ms) [inline src/vmware/azext_vmware/_params.py:47]: May I ask why do you want to deprecate the `--mi-system-assigned` parameter and redirect it to `--identity`? Please note we need to follow the [managed_identity_command_guideline](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md) to design parameters related to managed identities

Comment (cataggar) [reply src/vmware/azext_vmware/_params.py:47]: I did not realize that was the standard. Sorry @jonathanhe-msft for the busywork. Please undo some of these changes. It looks like the standard was being followed.

File: src/vmware/HISTORY.md [hunk 5]
```diff
@@ -1,5 +1,16 @@
 # Release History
 
+## 4.1.0 (2022-07)
+
+- Add new parameter `--identity` to `az vmware private-cloud create`
```
Comment (cataggar) [inline src/vmware/HISTORY.md:5]: undo

File: src/vmware/HISTORY.md [hunk 6]
```diff
@@ -1,5 +1,16 @@
...
+- Add new parameter `--identity` to `az vmware private-cloud update`
+- Add `az vmware private-cloud enable-cmk-encryption`
+- Add `az vmware private-cloud disable-cmk-encryption`
+- Deprecate `az vmware private-cloud add-cmk-encryption`
+- Deprecate `az vmware private-cloud delete-cmk-encryption`
+- Deprecate `--mi-system-assigned` parameter from `az vmware private-cloud create`
```
Comment (cataggar) [inline src/vmware/HISTORY.md:11]: undo

File: src/vmware/HISTORY.md [hunk 7]
```diff
@@ -1,5 +1,16 @@
...
+- Deprecate `az vmware private-cloud identity` command group
```
Comment (cataggar) [inline src/vmware/HISTORY.md:12]: undo

**Review by cataggar — CHANGES_REQUESTED** [review]

**Review by cataggar — APPROVED** [review]
Good job following the [identity command guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md).

**Comment by jonathanhe-msft** [comment]
@zhoxing-ms can I get a review on this please? thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**